### ISSUE TITLE
TypeError: t.getStructure is not a function when calling classDeclaration.getStructure()

### ISSUE THREAD

**Issue by dmnsgn** [dsherret/ts-morph#844]
**Describe the bug**

Version: 7.1.3

I am trying to get the structure of all the classes of a simple Namespace but it triggers an error due to comments. 

**To Reproduce**

Here is a reduced case of the declaration file:

```ts
declare namespace MYNAMESPACE {
  class MyClass {
    constructor(renderer: MYNAMESPACE.Renderer);
    /**
     * A flag
     * @member {boolean}
     * @readonly
     */
    readonly isActive: boolean;
  }
}

declare module "MYNAMESPACE.js" {
  export = MYNAMESPACE;
}
```

and the application code:

```ts
import TsMorph from "ts-morph";

const getNamespaceClasses = (namespace) => {
  let classes = namespace.getClasses();

  const namespaces = namespace.getNamespaces();

  for (let namespace of namespaces) {
    classes = [...classes, ...getNamespaceClasses(namespace)];
  }

  return classes;
};

const generate = async () => {
  // Create project and loop through classes
  const project = new TsMorph.Project();
  const sourceFiles = project.addSourceFilesAtPaths(`./**/*{.d.ts,.ts}`);

  let classes = [];
  for (let file of sourceFiles) {
    // Get namespace classes
    const namespaces = file.getNamespaces();
    for (let namespace of namespaces) {
      classes = [...classes, ...getNamespaceClasses(namespace)];
    }
    console.log(classes[0].getStructure());
  }
};

generate();

```

**Expected behavior**

Get the structure without error.

It seem to happen in the JSDoc class getStructure function where `tags: this.getTags().map(t => t.getStructure()),` fails
```ts
(node:39373) UnhandledPromiseRejectionWarning: TypeError: t.getStructure is not a function
    at /Users/path/to/project/nodes-generator-types/node_modules/ts-morph/dist/ts-morph.js:14470:45
```

Removing the following comment make it work (or at least not trigger the error):
```ts
    /**
     * A flag
     * @member {boolean}
     * @readonly
     */
```

**Comment by dsherret** [comment]
This was failing for the same reason as #779 (the JSDocReadonlyTag wasn't implemented... see https://github.com/dsherret/ts-morph/blob/latest/packages/ts-morph/wrapped-nodes.md#not-exist).

I've fixed this by implementing that tag. Will be released momentarily.

**Comment by dmnsgn** [comment]
@dsherret thank you for this. Could it be possible to also update `JSDoc.getStructure` to prevent the error?
https://github.com/dsherret/ts-morph/blob/51cbbea3da79b54e50afd9bec0cfb43f8737d073/packages/ts-morph/src/compiler/ast/doc/JSDoc.ts#L228-L234

The above could check for getStructure existence and filter the tags if not:

```js
 getStructure(): JSDocStructure { 
     return callBaseGetStructure<JSDocSpecificStructure>(JSDocBase.prototype, this, { 
         kind: StructureKind.JSDoc, 
         description: this.getDescription(), 
         tags: this.getTags().map(t => t && t.getStructure ? t.getStructure() : null).filter(Boolean), 
     }); 
 } 
```

Checking the above with a file that contains an `@author` tag (not yet implemented), the error is avoided.

Let me know how that sounds and if you want me to open PR for it.

**Comment by dsherret** [comment]
@dmnsgn it would probably be best not to do that because that code would cause some people to lose some tags. It would be better to just ensure everything is implemented.

The remaining JSDocTags are:

* JSDocAuthorTag
* JSDocCallbackTag
* JSDocEnumTag
* JSDocImplementsTag

So it would be best to just implement those to ensure this error doesn't occur. I've opened #854

**Comment by dmnsgn** [comment]
Sure, here the deprecated tag is not implemented so getStructure fails:
```ts
declare namespace MYNAMESPACE {
  class MyClass {
    constructor(renderer: MYNAMESPACE.Renderer);
    /**
     * @deprecated since 5.3.0
     */
    readonly isActive: boolean;
  }
}

declare module "MYNAMESPACE.js" {
  export = MYNAMESPACE;
}
```

**Comment by dsherret** [comment]
@dmnsgn sorry about that! Should be fixed now in 8.1.0.

It was fixed, but then they added the deprecated tag in 4.0, so I missed that.
